app/javascript/bundles/App/containers/MessageInput.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+/**
+ *
+ * MessageInput
+ *
+ * @func
+ * @param {Object} props
+ * @param {String} props.messageText
+ * @param {Callback} props.postMessage
+ * @param {Callback} props.setMessageText
+ * @return {ReactComponent}
+ *
+ * */
+
+function MessageInput ({ messageText, postMessage, setMessageText  }) {
```

Comment:
Yeah, this is a point of confusion for me in regards to React. I don't quite understand with any real depth how to take advantage of lifecycle hooks. Especially when you add Redux to the mix. Maybe that'll be more clear when we start integrating more of the Rails/API stuff.

Personally, I try to avoid relying in the "this" keyword as much as possible because it's super confusing for most people. Using class syntax encourages that, and when you're three layers removed from your action creator/event handler and you've had to bind "this" twice, I don't really see the point. Reverting the code to a function literal is just my way of making things more explicit and deterring the temptation to rely on "this", but also making the code reasonable if you _have_ to use "this" since because of how it's scoped.

I see your point about boxing myself in, though. I think I'm used to writing code that older/unfamiliar developers have to read through, so I may take it a bit too far sometimes.